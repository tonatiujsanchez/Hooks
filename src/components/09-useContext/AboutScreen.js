import { useContext } from "react"
import { UserContext } from "./UserContex"


const AboutScreen = () => {

    const { user, setUser } = useContext(UserContext)

    return (
        <div>
            <h1>AboutScreen</h1>
            <hr />
            <div>
                <pre>
                    { JSON.stringify( user, null, 4 ) }
                </pre>
            </div>
            <div>
                <button
                    onClick={ ()=> setUser({}) } 
                    className="btn btn-danger">
                        Logout
                </button>
            </div>
        </div>
    )
}

export default AboutScreen