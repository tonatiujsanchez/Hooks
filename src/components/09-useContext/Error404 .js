

const Error404  = () => {
  return (
    <div>
        <h1>Error 404</h1>
        <h2>PÃ¡gina no encontrada</h2>
    </div>
  )
}

export default Error404 