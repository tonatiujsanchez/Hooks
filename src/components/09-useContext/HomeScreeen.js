import { useContext } from "react"
import { UserContext } from "./UserContex"


const HomeScreeen = () => {

  const { user } = useContext( UserContext )

  return (
    <div>
        <h1>HomeScreeen</h1>
        <hr/>
        <pre>{ JSON.stringify( user, null, 4 ) }</pre>
    </div>
  )
}

export default HomeScreeen