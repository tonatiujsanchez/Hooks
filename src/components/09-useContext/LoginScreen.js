import { useContext } from "react"
import { UserContext } from "./UserContex"


const LoginScreen = () => {
    
    const { setUser } = useContext(UserContext);
    
    const user = {
        id: 123,
        name: 'Brandon',
        email: 'brandon@gmail.com'
    }

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />
            <div>
                <button
                    onClick={ ()=> setUser( user ) } 
                    className="btn btn-primary">
                        Login
                </button>
            </div>
        </div>
    )
}

export default LoginScreen