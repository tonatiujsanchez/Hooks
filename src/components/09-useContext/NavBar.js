import { NavLink } from "react-router-dom"


const NavBar = () => {
  return (
    <nav className="navbar navbar-expand navbar-light d-flex justify-content-center justify-content-sm-between mb-3 flex-wrap">
        <NavLink className="navbar-brand fw-bold fs-4" to="/">UseContext</NavLink>
        <ul className="navbar-nav">
            <li className="nav-item"><NavLink className="nav-link fw-bold" to="/">Home</NavLink></li>
            <li className="nav-item"><NavLink className="nav-link fw-bold" to="/about">About</NavLink></li>
            <li className="nav-item"><NavLink className="nav-link fw-bold" to="/login">login</NavLink></li>
        </ul>
    </nav>
  )
}

export default NavBar