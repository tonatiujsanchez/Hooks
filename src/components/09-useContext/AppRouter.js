import {
    BrowserRouter,
    Routes,
    Route,
  } from "react-router-dom";

import AboutScreen from "./AboutScreen";
import Error404 from "./Error404 ";
import HomeScreeen from "./HomeScreeen";
import LoginScreen from "./LoginScreen";


import NavBar from "./NavBar";

const AppRouter = () => {
  return (
    <BrowserRouter>
        <NavBar />
        <Routes>
            <Route path="/" element={ <HomeScreeen /> } />
            <Route path="/about" element={ <AboutScreen /> } />
            <Route path="/login" element={ <LoginScreen /> } />
            <Route path="*"  element={ <Error404 /> } />
        </Routes>
    </BrowserRouter>
  )
}

export default AppRouter