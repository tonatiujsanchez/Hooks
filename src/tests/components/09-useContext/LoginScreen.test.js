import { mount } from "enzyme"
import LoginScreen from "../../../components/09-useContext/LoginScreen"
import { UserContext } from "../../../components/09-useContext/UserContex"



describe('Pruebas en <LoginScreen/>', () => {
    const user = {
        id: 123,
        name: 'Brandon',
        email: 'brandon@gmail.com'
    }
    const setUser = jest.fn()
    const wrapper = mount(
        <UserContext.Provider value={{
            user,
            setUser
        }}>
            <LoginScreen />
        </UserContext.Provider>
    )



    test('Mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })



    test('Debe de ejecutar el setUser con el argumento esperado', () => {

        const buttonClick = wrapper.find('button').prop('onClick')
        buttonClick() 

        expect( setUser ).toBeCalledTimes(1)
        expect( setUser ).toHaveBeenCalledWith(user)


    })

})