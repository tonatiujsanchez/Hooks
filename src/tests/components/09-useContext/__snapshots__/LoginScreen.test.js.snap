// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen/> Mostrarse correctamente 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <div>
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
